reducers/gotPlayer/player.js

```diff
@@ -1,57 +1,96 @@
 import { playerActions } from '../../actions';
-// const byId = (state = {}, action) => {
-//   switch (action.type) {
-//     case expression:
-//
-//       break;
-//     default:
-//
-//   }
-// };
+import { combineReducers } from 'redux';
 
 const initialState = {
   loading: true,
   error: false,
+  loaded: false,
   player: {
     profile: {},
   },
 };
 
-export default (state = initialState, action) => {
+const player = (state = initialState, action) => {
   switch (action.type) {
     case playerActions.REQUEST:
       return {
         ...state,
         loading: true,
+        loaded: false,
       };
     case playerActions.OK:
       return {
         ...state,
         loading: false,
+        loaded: true,
         player: action.payload,
       };
     case playerActions.ERROR:
       return {
         ...state,
         loading: false,
         error: true,
+        loaded: false,
       };
     default:
       return state;
   }
 };
 
+const byId = (state = {}, action) => {
+  switch (action.type) {
+    case playerActions.REQUEST:
+    case playerActions.OK:
+    case playerActions.ERROR:
+      return {
+        ...state,
+        [action.id]: player(state[action.id], action),
+      };
+    default:
+      return state;
+  }
+};
+
+const allIds = (state = [], action) => {
+  switch (action.type) {
+    case playerActions.OK:
+      if (state.includes(action.id)) {
+        return state;
+      }
+      return [...state, action.id];
+    default:
+      return state;
+  }
+};
+
+export default combineReducers({
+  byId,
+  allIds,
+});
+
+
 export const getPlayer = {
-  getError: state => state.yaspReducer.gotPlayer.playerReducer.error,
-  getLoading: state => state.yaspReducer.gotPlayer.playerReducer.loading,
-  getPlayer: state => state.yaspReducer.gotPlayer.playerReducer.player,
-  getProfile: state => getPlayer.getPlayer(state).profile,
-  getPlayerName: state => getPlayer.getPlayer(state).profile.personaname,
-  getLastLogin: state => getPlayer.getPlayer(state).profile.last_login,
-  getMmrEstimate: state => getPlayer.getPlayer(state).mmr_estimate,
-  getSoloMmrEstimate: state => getPlayer.getPlayer(state).solo_competitive_rank,
-  getCompetitiveRank: state => getPlayer.getPlayer(state).competitive_rank,
-  getPicture: state => getPlayer.getProfile(state).avatarmedium,
-  getSteamLink: state => getPlayer.getProfile(state).profileurl,
-  getCheese: state => getPlayer.getProfile(state).cheese,
+  getPlayerById: (state, id) => {
+    // this feels like a hack
+    if (!state.yaspReducer.gotPlayer.playerReducer.byId[id]) {
```

Comment:
explain why it's a hack.

Also, isn't it unsafe to access deeply nested properties like this?  If any of them are undefined, the application will crash.